experanto/datasets.py

```diff
@@ -394,59 +436,189 @@ def initialize_transforms(self):
+        # Check if it's already a callable (function)
+        if callable(filter_config):
+            # print(f"DEBUG: callable(filter_config) returned True for type {type(filter_config)}. Returning config directly.")
+            return filter_config
+        
+        # Check if it's a config that needs instantiation
+        if isinstance(filter_config, (dict, DictConfig)) and '__target__' in filter_config:
+            try:
+                # Manually handle instantiation for factory pattern with __partial__=True
+                target_str = filter_config['__target__']
+                module_path, func_name = target_str.rsplit('.', 1)
+                
+                # Import the module and get the factory function
+                module = importlib.import_module(module_path)
+                factory_func = getattr(module, func_name)
+                
+                # Prepare arguments for the factory function (excluding special keys)
+                args = {k: v for k, v in filter_config.items() if k not in ('__target__', '__partial__')}
+                
+                # Call the factory function with its arguments to get the actual implementation function
+                implementation_func = factory_func(**args)
+                return implementation_func
+                
+            except (ImportError, AttributeError, KeyError, TypeError) as e:
+                raise TypeError(f"Failed to manually instantiate filter from config {filter_config}: {e}")
+            
+        raise TypeError(f"Filter config must be either callable or a valid config dict with __target__, got {type(filter_config)}")
+
+    def get_valid_intervals_from_filters(self, visualize: bool = False) -> List[TimeInterval]:
+        valid_intervals = None
+        for modality in self.modality_config:
+            if "filters" in self.modality_config[modality]:
+                device = self._experiment.devices[modality]
+                for filter_name, filter_config in self.modality_config[modality]["filters"].items():
+                    # Get the final callable filter function
+                    filter_function = self._get_callable_filter(filter_config)
+                    valid_intervals_ = filter_function(device_=device)
+                    if visualize:
+                        print(f"modality: {modality}, filter: {filter_name}")
+                        visualization_string = get_stats_for_valid_interval(valid_intervals_, self.start_time, self.end_time)
+                        print(visualization_string)
+                    if valid_intervals is None:
+                        valid_intervals = valid_intervals_
+                    else:
+                        valid_intervals = find_intersection_between_two_interval_arrays(valid_intervals, valid_intervals_)
+
+        return valid_intervals
+    
+        
+    def get_condition_mask_from_meta_conditions(self, valid_conditions_sum_of_product: List[dict]) -> np.ndarray:
```

Comment:
I feel we should also compare how they do it in the temporal data of the Dyer-Lab.

Reply:
That's not required right now but a comment for later.